#!/usr/bin/env python
PACKAGE = "thormang3_walk_control_module"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

####### Balance enable #####
gen.add("balance",   bool_t,   0, "Balance on/off",  True)

####### cob_offset #######
gen.add("cob_x_offset_m",                   double_t, 0, "", -0.01, -0.1, 0.1)
gen.add("cob_y_offset_m",                   double_t, 0, "",  0,    -0.1, 0.1)

####### FeedForward #####
gen.add("hip_roll_swap_angle_rad",          double_t, 0, "",  0.008, 0,   0.02)

########## Gain ########
# gyro
gen.add("gyro_gain",                        double_t, 0, "",  0.6,   0,   2)

# imu
gen.add("foot_roll_angle_gain",             double_t, 0, "", -1,   -10,   0)
gen.add("foot_pitch_angle_gain",            double_t, 0, "", -1,   -10,   0)

#ft sensor
gen.add("foot_x_force_gain",                double_t, 0, "",  0.1,   0,   1)
gen.add("foot_y_force_gain",                double_t, 0, "",  0.1,   0,   1)
gen.add("foot_z_force_gain",                double_t, 0, "", -0.025, -0.1, 0)
gen.add("foot_roll_torque_gain",            double_t, 0, "",  0.0025, 0,   0.01)
gen.add("foot_pitch_torque_gain",           double_t, 0, "",  0.0015,0,   0.01)

########## TIME CONSTANT ##########
# imu
gen.add("foot_roll_angle_time_constant",    double_t, 0, "",  0.05,   0,   1)
gen.add("foot_pitch_angle_time_constant",   double_t, 0, "",  0.05,   0,   1)

#ft
gen.add("foot_x_force_time_constant",       double_t, 0, "",  0.2,   0,   1)
gen.add("foot_y_force_time_constant",       double_t, 0, "",  0.2,   0,   1)
gen.add("foot_z_force_time_constant",       double_t, 0, "",  0.1,   0,   1)
gen.add("foot_roll_torque_time_constant",   double_t, 0, "",  0.1,   0,   1)
gen.add("foot_pitch_torque_time_constant",  double_t, 0, "",  0.1,   0,   1)

exit(gen.generate(PACKAGE, "ROBOTIS", "BalanceParameters"))
